maubot/handlers/command.py

```diff
@@ -116,9 +121,16 @@ async def __call__(self, evt: MaubotMessageEvent, *, _existing_args: Dict[str, A
             await evt.reply(self.__mb_full_help__)
             return
 
-        if self.__bound_instance__:
-            return await self.__mb_func__(self.__bound_instance__, evt, **call_args)
-        return await self.__mb_func__(evt, **call_args)
+        try:
+            if self.__bound_instance__:
+                return await self.__mb_func__(self.__bound_instance__, evt, **call_args)
+            return await self.__mb_func__(evt, **call_args)
+        except CommandError as e:
+            await evt.reply(f"Error: {e}")
+            raise e
+        except Exception as e:
+            await evt.reply("An error happened while running the command")
+            raise e
```

Comment:
This looks like it's going to reply multiple times if a subcommand fails, might need to check that it's the top-level command before replying

Reply:
I’m not sure i understand how. After a simple check, it doesn’t seem that the toplevel command code is called when i call a subcommand

Reply:
Oh right, that goes through a different path, probably fine then